frame/system/src/tests.rs

```diff
@@ -59,27 +59,43 @@ fn stored_map_works() {
 	});
 }
 
+#[test]
+fn inc_account_nonce_works() {
+	new_test_ext().execute_with(|| {
+		System::set_block_number(2);
+		System::inc_account_nonce(&0);
+		//initial call sets nonce to the block number
+		assert_eq!(System::account_nonce(&0), 2);
+		//subsequent calls increment nonce
+		System::inc_account_nonce(&0);
+		assert_eq!(System::account_nonce(&0), 3);
+		System::inc_account_nonce(&0);
+		assert_eq!(System::account_nonce(&0), 4);
+	});
+}
+
 #[test]
 fn provider_ref_handover_to_self_sufficient_ref_works() {
 	new_test_ext().execute_with(|| {
 		assert_eq!(System::inc_providers(&0), IncRefStatus::Created);
+		System::set_block_number(2);
```

Comment:
Why have you added this line and updated all the nonces below?

Reply:
In the original test, the block number would be 0, and so nonce will be set to 0 even after inc_account_nonce is called. Felt that this wouldn't be helpful behavior for the subsequent tests, especially since one of them tests what will happen to nonce if you delete the account. Changed it to 2 (any non zero number would work) so that the tests would behave similarly as before.

Reply:
actually could also set the block number to 1, not sure why i used 2

Reply:
yeah, it might also make sense to handle when block number is zero, and increment nonce to 1. You can think about the most concise logic to handle this stuff.

Reply:
yea thinking through it should probably handle the block number = 0 case, even if realistically i think that wouldn't occur right

Reply:
It could occur if a user sets up a special genesis which would trigger transactions on the genesis block or something.